Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx

```diff
@@ -72,6 +72,8 @@ const SUMMARY_ARROW_SPACE = 28; // the rough pixel size of the dropdown arrow to
 
 // -------------------- Helper functions -------------------- //
 
+type GetTriggerIndex = (trigger: ITrigger) => number;
```

Comment:
how about a util function that takes a dialog and a trigger and it return the trigger's index.